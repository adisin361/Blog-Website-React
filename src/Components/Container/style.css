.container {
  display: flex;
  padding-top: 2vh;
  padding-left: 19vw;
  padding-right: 19vw;
  justify-content: space-between;
  flex-wrap: wrap;
  align-items: center;
}