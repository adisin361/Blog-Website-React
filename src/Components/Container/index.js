import * as React from 'react';
import './style.css';

import Card from '../Card';
import { GET_BLOG_DATA } from '../../Constants/apiEndPoints';
import makeRequest from '../../utils/makeRequest';

const Container = () => {

  const [blogData, setBlogData] = React.useState();
  const [error, setError] = React.useState();

  React.useEffect(() => {
    makeRequest(GET_BLOG_DATA)
      .then((response) => {
        setBlogData(response);
      })
      .catch((e) => {
        setError(e.message);
      });
  }, []);

  if (error) {
    return (
      <div className='blogDataError'>
        <p>{error}</p>
      </div>
    );
  }

  return blogData ? (
    <>
      <div className='container'>
        {blogData.map((eachBlogData) =>
        (
          <Card key={eachBlogData.id} blogData={eachBlogData} />
        ))}
      </div>
    </>
  ) : (
    <div className='blogDataLoader'>
      <p>Loading...</p>
    </div>
  );
};

export default Container;