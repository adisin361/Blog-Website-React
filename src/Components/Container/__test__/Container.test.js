import "@testing-library/jest-dom";
import { fireEvent, render } from "@testing-library/react";
import React from "react";
import Container from '..';

describe("Container", () => {
  // it("should match the snapshot", () => {
  //   const { asFragment } = render(<Container />);
  //   expect(asFragment()).toMatchSnapshot();
  // });
});