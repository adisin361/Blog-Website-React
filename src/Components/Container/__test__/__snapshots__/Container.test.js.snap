// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div>
      <div
        class="card"
      >
        <div
          class="image"
        >
          <img
            alt="abstract"
            src="abstract.png"
          />
        </div>
        <div
          class="description-section"
        >
          <div>
            <div
              class="date-time"
            >
              <p>
                2nd Januray, 2018
              </p>
              <p>
                2 mins
              </p>
            </div>
            <p
              class="heading"
            >
              The future of abstract art and the culture ...
            </p>
            <p
              class="meta-data"
            >
              Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
            </p>
          </div>
          <div
            class="footer"
          >
            <center>
              <hr />
            </center>
            <div
              class="reaction"
            >
              <div
                class="clap"
              >
                <img
                  alt="clapIcon"
                  src="clapping.svg"
                />
                <p>
                  10
                </p>
              </div>
              <img
                alt="heartIcon"
                src="heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="card"
      >
        <div
          class="image"
        >
          <img
            alt="abstract"
            src="outdoor.png"
          />
        </div>
        <div
          class="description-section"
        >
          <div>
            <div
              class="date-time"
            >
              <p>
                31st Januray, 2018
              </p>
              <p>
                1 min
              </p>
            </div>
            <p
              class="heading"
            >
              How painting outdoors can transform your 
            </p>
            <p
              class="meta-data"
            >
              Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
            </p>
          </div>
          <div
            class="footer"
          >
            <center>
              <hr />
            </center>
            <div
              class="reaction"
            >
              <div
                class="clap"
              >
                <img
                  alt="clapIcon"
                  src="clapping.svg"
                />
                <p>
                  1
                </p>
              </div>
              <img
                alt="heartIcon"
                src="heart-red.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="card"
      >
        <div
          class="image"
        >
          <img
            alt="abstract"
            src="young-painters.png"
          />
        </div>
        <div
          class="description-section"
        >
          <div>
            <div
              class="date-time"
            >
              <p>
                20th March, 2018
              </p>
              <p>
                2 mins
              </p>
            </div>
            <p
              class="heading"
            >
              10 young painters you need to know
            </p>
            <p
              class="meta-data"
            >
              Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
            </p>
          </div>
          <div
            class="footer"
          >
            <center>
              <hr />
            </center>
            <div
              class="reaction"
            >
              <div
                class="clap"
              >
                <img
                  alt="clapIcon"
                  src="clapping.svg"
                />
                <p>
                  15
                </p>
              </div>
              <img
                alt="heartIcon"
                src="heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="card"
      >
        <div
          class="image"
        >
          <img
            alt="abstract"
            src="perfect-art.png"
          />
        </div>
        <div
          class="description-section"
        >
          <div>
            <div
              class="date-time"
            >
              <p>
                22nd June, 2018
              </p>
              <p>
                5 mins
              </p>
            </div>
            <p
              class="heading"
            >
              Tips for creating the perfect art
            </p>
            <p
              class="meta-data"
            >
              Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
            </p>
          </div>
          <div
            class="footer"
          >
            <center>
              <hr />
            </center>
            <div
              class="reaction"
            >
              <div
                class="clap"
              >
                <img
                  alt="clapIcon"
                  src="clapping.svg"
                />
                <p>
                  11
                </p>
              </div>
              <img
                alt="heartIcon"
                src="heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="card"
      >
        <div
          class="image"
        >
          <img
            alt="abstract"
            src="inspired.png"
          />
        </div>
        <div
          class="description-section"
        >
          <div>
            <div
              class="date-time"
            >
              <p>
                4th Feburary, 2019
              </p>
              <p>
                9 mins
              </p>
            </div>
            <p
              class="heading"
            >
              Stay inspired: top artists blog ...
            </p>
            <p
              class="meta-data"
            >
              Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
            </p>
          </div>
          <div
            class="footer"
          >
            <center>
              <hr />
            </center>
            <div
              class="reaction"
            >
              <div
                class="clap"
              >
                <img
                  alt="clapIcon"
                  src="clapping.svg"
                />
                <p>
                  45
                </p>
              </div>
              <img
                alt="heartIcon"
                src="heart-red.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="card"
      >
        <div
          class="image"
        >
          <img
            alt="abstract"
            src="reason.png"
          />
        </div>
        <div
          class="description-section"
        >
          <div>
            <div
              class="date-time"
            >
              <p>
                10th Januray, 2019
              </p>
              <p>
                2 mins
              </p>
            </div>
            <p
              class="heading"
            >
              5 reasons why everyone should try an...
            </p>
            <p
              class="meta-data"
            >
              Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
            </p>
          </div>
          <div
            class="footer"
          >
            <center>
              <hr />
            </center>
            <div
              class="reaction"
            >
              <div
                class="clap"
              >
                <img
                  alt="clapIcon"
                  src="clapping.svg"
                />
                <p>
                  11
                </p>
              </div>
              <img
                alt="heartIcon"
                src="heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
